### PR TITLE
NAVBAR ADDED

### PR DESCRIPTION
**Pull request by RohithChittibommala**
I ADDED NAVBAR HOWEVER THERE ARE SOME BUGS I NEED SOME GUIDANCE REGARDING THIS

### CODE REVIEW

**Comment by pAditya198** [comment]
@anujpuri72 don't merge the PR squash and merge it. After we have reviewed it.

**Comment by pAditya198** [comment]
@RohithChittibommala  add screenshots of the updates

File: client/src/Components/App.js [hunk 1]
```diff
@@ -2,13 +2,11 @@ import React, { Component } from "react";
...
+import NavBar from "./Home/NavBaar";
+import ContactUsForm from "./Home/contactUs";
 class App extends Component {
   render() {
-    return (
-      <Switch>
-        <Route exact path="/" component={Home} />
-      </Switch>
-    );
+    return <Home />;
```
Comment (pAditya198) [inline client/src/Components/App.js]: Why did you removed the route? Change it back

File: client/src/Components/CusButton.jsx [hunk 2]
```diff
@@ -1,31 +1,62 @@
...
+    target.classList.add("drag");
+    console.log(target);
     e.dataTransfer.setData("card_id", target.id);
     setTimeout(() => {
       target.style.display = "none";
     }, 0);
   };
+  handleInputChange = (e) => {
+    const currentTarget = e.currentTarget;
+    const value = currentTarget.value;
```
Comment (pAditya198) [inline client/src/Components/CusButton.jsx:17]: Is there really a requirement to add two constants to set state?

File: client/src/Components/CusButton.jsx [hunk 3]
```diff
@@ -1,31 +1,62 @@
...
+    this.setState({ value });
+  };
   dragover = (e) => {
     e.stopPropagation();
   };
+  getInputStyles = (color, radius) => {
+    console.log("get Input styles" + color);
```
Comment (pAditya198) [inline client/src/Components/CusButton.jsx]: remove console.log()

File: client/src/Components/CusButton.jsx [hunk 4]
```diff
@@ -1,31 +1,62 @@
...
+    const styles = {};
+    styles.color = `white`;
+    styles.display = "block";
+    styles.borderRadius = `${radius}`;
+    styles.backgroundColor = `${color}`;
```
Comment (pAditya198) [inline client/src/Components/CusButton.jsx:33]: you are not reusing the object styles again so avoid creating a const and simply 
return {
objectData
}

File: client/src/Components/CusButton.jsx [hunk 5]
```diff
@@ -1,31 +1,62 @@
...
+        style={this.getButtonStyles()}
       >
-        hello
+        <input
+          type="text"
+          style={this.getInputStyles(backgroundColor, borderRadius)}
+          value={this.state.value}
+          className="buttonInput"
+          onChange={(e) => this.handleInputChange(e)}
+        />
```
Comment (pAditya198) [inline client/src/Components/CusButton.jsx:63]: Why did you add input field inside a button?

File: client/src/Components/Home/DarkNavBar.jsx [hunk 6]
```diff
@@ -0,0 +1,100 @@
+import React, { Component } from "react";
+// import "../../scss/components/_navbar.scss";
```
Comment (pAditya198) [inline client/src/Components/Home/DarkNavBar.jsx]: remove comment if not necessary

File: client/src/Components/Home/DarkNavBar.jsx [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+console.log("hell");
```
Comment (pAditya198) [inline client/src/Components/Home/DarkNavBar.jsx]: remove console.log()

File: client/src/Components/Home/DarkNavBar.jsx [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+class DarkNav extends Component {
+  state = {
+    link1: "Home",
+    link2: "Design",
+    link3: "Contact us",
+    link4: "Share",
+    backgroundColor: "",
+  };
+  componentDidMount() {
+    const backgroundColor = this.props.styles.backgroundColor;
```
Comment (pAditya198) [inline client/src/Components/Home/DarkNavBar.jsx:11]: was the variable necessary

File: client/src/Components/Home/DarkNavBar.jsx [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+    this.setState({ backgroundColor });
+  }
+  handleInputChange = (e, label) => {
+    const value = e.currentTarget.value;
+    console.log(label);
```
Comment (pAditya198) [inline client/src/Components/Home/DarkNavBar.jsx]: remove log

File: client/src/Components/Home/DarkNavBar.jsx [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+  dragover = (e) => {
+    const target = e.target;
+    // target.classList.remove("drag");
+    console.log(target);
+    e.stopPropagation();
+  };
+  getNavbarStyles = () => {
+    const element = this.props.styles;
+    console.log(element);
+    console.log("get " + element.height);
```
Comment (pAditya198) [inline client/src/Components/Home/DarkNavBar.jsx]: remove log

File: client/src/Components/Home/DarkNavBar.jsx [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+  };
+  // renderInput = (property) => {
+  //   return (
+  //     <input
+  //       type="text"
+  //       id={`${property}`}
+  //       placeholder={`Enter ${propertygit}`}
+  //     />
+  //   );
+  // };
```
Comment (pAditya198) [inline client/src/Components/Home/DarkNavBar.jsx]: remove comments

File: client/src/Components/Home/DarkNavBar.jsx [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+                type="text"
+                value={this.state[m]}
+                style={this.getInputStyles()}
+                onChange={(e) => this.handleInputChange(e, `${m}`)}
+                className="anchorInput"
+              />
+            </a>
+          ))}
+        </div>
+      </React.Fragment>
```
Comment (pAditya198) [inline client/src/Components/Home/DarkNavBar.jsx:85]: There is no need of React Fragment here.

File: client/src/Components/Home/NavBaar.jsx [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+          {this.renderNavbar("grey", "18px", "nav1", "add navbar")}
+        </div>
+        <div
+          id="dropspace"
+          onDrop={this.onDrop}
+          onDragOver={this.dragOver}
+          onClick={this.getPositions}
+        >
+          {elements.map((element) => {
+            console.log(element);
```
Comment (pAditya198) [inline client/src/Components/Home/NavBaar.jsx]: remove log

File: client/src/Components/Home/NavBaar.jsx [hunk 14]
```diff
@@ -0,0 +1,97 @@
...
+      element.id = id;
+      this.setState({ didClick: false });
+    } else {
+      return;
+    }
+    elements.push(element);
+  };
+  render() {
+    const { elements } = this.state;
+    console.log(elements);
```
Comment (pAditya198) [inline client/src/Components/Home/NavBaar.jsx]: remove log

File: client/src/Components/Home/NavBaar.jsx [hunk 15]
```diff
@@ -0,0 +1,97 @@
...
+    navbar.style.display = "block";
+    navbar.style.top = `${_mouseY}px`;
+    navbar.style.left = `${_mouseX - 200}px `;
+    // console.log(card.style.top, card.style.top);
+  };
+  dragOver = (e) => {
+    e.preventDefault();
+  };
+  getPositions = (ev) => {
+    console.log("hello");
```
Comment (pAditya198) [inline client/src/Components/Home/NavBaar.jsx]: remove log

File: client/src/Components/Home/NavBaar.jsx [hunk 16]
```diff
@@ -0,0 +1,97 @@
...
+    e.preventDefault();
+    const _mouseY = e.clientY;
+    const _mouseX = e.clientX;
+    const navbar_id = e.dataTransfer.getData("navbar_id");
+    console.log(navbar_id);
+    const navbar = document.getElementById(navbar_id);
+    navbar.style.display = "block";
+    navbar.style.top = `${_mouseY}px`;
+    navbar.style.left = `${_mouseX - 200}px `;
+    // console.log(card.style.top, card.style.top);
```
Comment (pAditya198) [inline client/src/Components/Home/NavBaar.jsx]: remove comment

File: client/src/Components/Home/NavBaar.jsx [hunk 17]
```diff
@@ -0,0 +1,97 @@
...
+
+  setNavbarProperties = (backgroundColor, borderRadius, id) => {
+    this.setState({ backgroundColor, borderRadius, id, didClick: true });
+  };
+  onDrop = (e) => {
+    e.preventDefault();
+    const _mouseY = e.clientY;
+    const _mouseX = e.clientX;
+    const navbar_id = e.dataTransfer.getData("navbar_id");
+    console.log(navbar_id);
```
Comment (pAditya198) [inline client/src/Components/Home/NavBaar.jsx]: remove log

File: client/src/Components/Home/NavBaar.jsx [hunk 18]
```diff
@@ -0,0 +1,97 @@
...
+class NavBar extends Component {
+  state = {
+    backgroundColor: "",
+    borderRadius: "",
+    didClick: false,
+    id: "",
+    elements: [],
+  };
+  renderNavbar = (color, radius, id, label) => {
+    console.log("render Navbar");
```
Comment (pAditya198) [inline client/src/Components/Home/NavBaar.jsx]: remove log

File: client/src/scss/components/_aside.scss [hunk 19]
```diff
@@ -1,13 +1,45 @@
...
+  border-right: 1px dashed #000;
   flex-flow: column;
 }
-.Home aside button {
-  width: 150px;
-  margin: 10px auto;
+// .Home aside button {
+//   width: 150px;
+//   margin: 10px auto;
+// }
```
Comment (pAditya198) [inline client/src/scss/components/_aside.scss]: remove comment

File: client/src/scss/components/_aside.scss [hunk 20]
```diff
@@ -1,13 +1,45 @@
+@import url("https://fonts.googleapis.com/css2?family=Didact+Gothic&display=swap");
+// @import url("https://fonts.googleapis.com/css2?family=Didact+Gothic&display=swap");
```
Comment (pAditya198) [inline client/src/scss/components/_aside.scss]: remove comment

File: client/src/scss/components/_navbar.scss [hunk 21]
```diff
@@ -0,0 +1,62 @@
...
+}
+.ome {
+  width: 100px;
+  display: inline-block;
+  // display: flexbox;
+}
+// .ome .button {
+//   width: 70px;
+//   height: 80px;
+// }
```
Comment (pAditya198) [inline client/src/scss/components/_navbar.scss]: remove comment

**Review by pAditya198 — CHANGES_REQUESTED** [review]

**Comment by anujpuri72** [comment]
@RohithChittibommala did u broke side menu functionality on right clicking the element???

**Review by anujpuri72 — CHANGES_REQUESTED** [review]
U removed on the onContext menu and toggle menu  methods from all the components . check the 'Dropdown'  pr for the implementation

**Comment by pAditya198** [comment]
@anujpuri72 everything seems fine squash and this PR

**Pull request merged**